### PR TITLE
First stab at updates for Source, APIs, Search Platform

### PR DESCRIPTION
**Pull request by mrnugget**
This updates the `.yml` files so that all the relationships and roles are up-to-date.

It also updates the team pages - at least somewhat. The "Search Core" team page still needs to be moved/renamed, but I didn't want to change too many links.

Still need to do more revisions, but I don't want to add too much into a single PR.

### CODE REVIEW

File: content/strategy-goals/strategy/platform/search_platform.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Search Platform strategy
```
Comment (stefanhengl) [inline content/strategy-goals/strategy/platform/search_platform.md:1]: This page seems very outdated. Apart from the link at the top we should probably start from scratch?

Comment (mrnugget) [reply content/strategy-goals/strategy/platform/search_platform.md:1]: Alright! Sounds good

Comment (mrnugget) [reply content/strategy-goals/strategy/platform/search_platform.md:1]: Done!

File: data/team.yml [hunk 2]
```diff
@@ -83,15 +83,39 @@ joe_chen:
...
+  pronouns: he/him/his
+  role: Engineering Manager, APIs
+  reports_to: eng_lead
+  manager_role_slug: apis_lead
+  hide_on_team_page: true
+
+thorsten_ball_search_platform:
+  name: Thorsten Ball
+  pronouns: he/him/his
+  role: Engineering Manager, Search Platform (interims)
```
Comment (BolajiOlajide) [inline data/team.yml]:
```suggestion
  role: Engineering Manager, Search Platform (interim)
```

I think this should be `interim`.

**Review by BolajiOlajide — APPROVED** [review]
LGTM

**Review by keegancsmith — APPROVED** [review]
#managerswritingcodekinda

**Review by efritz — APPROVED** [review]

**Pull request merged**